pages/blog/generated-columns-in-crate.html

```diff
@@ -0,0 +1,188 @@
+
+For our log data and analyzing queries use case, the most suitable method would be to partition the table by a column containing a timestamp. But what if the column in the [partitioned by](https://crate.io/docs/en/latest/sql/reference/create_table.html#partitioned-by-clause) clause has low granularity or does not suit our queries well enough. In both cases we would end up either with too many partitions in the table or find ourselves performing additional work to adopt the queries.
+
+With _generated columns_ the challenge can be addressed simply. Let's create an example of analyzing data logs on a monthly basis, while having day granularity for the timestamped column.
+
+First, create a `log` table with the new generated column `month` and put it into [partitioned by](https://crate.io/docs/en/latest/sql/reference/create_table.html#partitioned-by-clause) clause
+
+```bash
+cr> create table logs(
+      log string,
+      created timestamp,
+      month as date_trunc('month', created)
+    ) partitioned by (month);
+CREATE OK (... sec)
+```
+
+In the next step, we insert log records into the table.
+
+```bash
+cr> insert into logs (log, created) values
+      ('log_2015_01_01', '2015-01-01'),
+      ('log_2015-01_02', '2015-01-02'),
+      ('log_2015_02_01', '2015-02-01'),
+      ('log_2015_02_02', '2015-02-02'),
+      ('log_2015_03_01', '2015-03-01');
+INSERT OK, 5 row affected (... sec)
+```
+
+We can see the table has been partitioned by the generated `month` column.
+
+```bash
+cr> select table_name, partition_ident, 'values'
+      from information_schema.table_partitions
+      WHERE table_name = 'logs';
++------------+--------------------------+--------------------------+
+| table_name | partition_ident          | values                   |
++------------+--------------------------+--------------------------+
+| log        | 04732d1i6kojce1g60o30c1g | {'month': 1425168000000} |
+| log        | 04732d1i60o3ec1k60o30c1g | {'month': 1420070400000} |
+| log        | 04732d1i68rj8e1o60o30c1g | {'month': 1422748800000} |
++------------+--------------------------+--------------------------+
+SELECT 3 rows in set (... sec)
+```
+
+Looks good! Now we can analyze the log data on a monthly basis and utilize all the benefits of having a partitioned table. As you can see the combination of [partitioned tables](https://crate.io/docs/en/latest/sql/partitioned_tables.html) and [generated columns](https://crate.io/docs/en/latest/sql/generated-columns.html) provides a great tool in the Crate toolkit.
+
+### Query optimization on generated column partitions
+The implementation of _generated columns_ for the [partitioned by](https://crate.io/docs/en/latest/sql/reference/create_table.html#partitioned-by-clause) clause has been optimized for special cases.
+
+Selection on _generated columns_ uses current optimization.
```

Comment:
@kovrus I'm not really sure what you're trying to say here? Is it needed or maybe you can tell me further what you're trying to say?